eth_portfolio/structs.py

```diff
@@ -118,11 +118,16 @@ def __init_subclass__(cls, **kwargs):
         # Use the class name exactly as it is defined (e.g., TokenTransfer)
         cls_name = cls.__name__
 
+        # Replace {cls_name} in the class-level docstring
+        if cls.__doc__:
+            cls.__doc__ = cls.__doc__.replace("{cls_name}", cls_name)
+
         # Replace {cls_name} in attribute-level docstrings
-        for key in cls.__annotations__:
+        for key, attr_type in cls.__annotations__.items():
             attr = getattr(cls, key, None)
-            if attr is not None and attr.__doc__:
-                attr.__doc__ = attr.__doc__.format(cls_name=cls_name)
+            if attr and isinstance(attr, str):  # Ensure that attr is a docstring
```

Comment:
attr will be a function or an attribute, attr.__doc__ would be the docstring for the attr

Reply:
`attr.__doc__` I mean

Reply:
see if it works with updated by Cursor

Reply:
lol what?